publish/pub.go

```diff
@@ -40,7 +40,7 @@ type Reporter func(context.Context, PendingReq) error
 // queue size. As the publisher is not waiting for the outputs ACK, the total
 // number requests(events) active in the system can exceed the queue size. Only
 // the number of concurrent HTTP requests trying to publish at the same time is limited.
-type publisher struct {
+type Publisher struct {
```

Comment:
why is this public?

Reply:
Because it is returned in [NewPublisher](https://github.com/elastic/apm-server/pull/2301/files/7472c1f587def4e3b62d8d0a8964d8e05602e349#diff-f2dc32e949f35561380c5c85bc282679R73) which is a public method. Returned types should also be public.